### PR TITLE
Fixing System.Diagnostics.Tests.ProcessTests failures on Unix

### LINKED ISSUE #27459: System.Diagnostics.Tests.ProcessTests failures on Unix
System.Diagnostics.Tests.ProcessTests.HandleCountChanges fails with:

```
MESSAGE:
Assert.All() Failure: 1 out of 1 items in the collection did not pass.
[0]: Xunit.Sdk.TrueException: Field marshaling is not supported by RemoteInvoke: s_allowedProgramsToRun
Expected: True
Actual: False
 at Xunit.Assert.True(Nullable`1 condition, String userMessage)
 at System.Diagnostics.RemoteExecutorTestBase.<>c.<GetMethodInfo>b__14_0(FieldInfo fi) in /mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/_/fx/src/CoreFx.Private.TestUtilities/src/System/Diagnostics/RemoteExecutorTestBase.cs:line 180
 at Xunit.Assert.All[T](IEnumerable`1 collection, Action`1 action)
+++++++++++++++++++
STACK TRACE:
at System.Diagnostics.RemoteExecutorTestBase.GetMethodInfo(Delegate d) in /mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/_/fx/src/CoreFx.Private.TestUtilities/src/System/Diagnostics/RemoteExecutorTestBase.cs:line 178 at System.Diagnostics.RemoteExecutorTestBase.RemoteInvoke(Func`1 method, RemoteInvokeOptions options) in /mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/_/fx/src/CoreFx.Private.TestUtilities/src/System/Diagnostics/RemoteExecutorTestBase.cs:line 46 at System.Diagnostics.Tests.ProcessTests.HandleCountChanges() in /mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/_/fx/src/System.Diagnostics.Process/tests/ProcessTests.cs:line 1289
```

System.Diagnostics.Tests.ProcessTests.GetProcesses_InvalidMachineName_ThrowsInvalidOperationException  fails with:

```
MESSAGE:
Assert.All() Failure: 1 out of 1 items in the collection did not pass.
[0]: Xunit.Sdk.TrueException: Field marshaling is not supported by RemoteInvoke: s_allowedProgramsToRun
 Expected: True
 Actual: False
 at Xunit.Assert.True(Nullable`1 condition, String userMessage)
 at System.Diagnostics.RemoteExecutorTestBase.<>c.<GetMethodInfo>b__14_0(FieldInfo fi) in /mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/_/fx/src/CoreFx.Private.TestUtilities/src/System/Diagnostics/RemoteExecutorTestBase.cs:line 180
 at Xunit.Assert.All[T](IEnumerable`1 collection, Action`1 action)
+++++++++++++++++++
STACK TRACE:
at System.Diagnostics.RemoteExecutorTestBase.GetMethodInfo(Delegate d) in /mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/_/fx/src/CoreFx.Private.TestUtilities/src/System/Diagnostics/RemoteExecutorTestBase.cs:line 178 at System.Diagnostics.RemoteExecutorTestBase.RemoteInvoke(Func`1 method, RemoteInvokeOptions options) in /mnt/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_ubuntu_corefx_baseline_prtest/_/fx/src/CoreFx.Private.TestUtilities/src
```

This failure was masked by https://github.com/dotnet/corefx/issues/27426

### PR DESCRIPTION
**Pull request by maryamariyan**
Making sure we throw InvalidOperationException and with proper message to be consistent with netfx.

This fixes test HandleCountChanges and updates two other existing tests

Making s_allowedProgramsToRun field static in order to fix the 'field marshalling' error occuring during RemoteInvoke

Fixes #27459 

cc: @wtgodbe @danmosemsft @jkotas

### CODE REVIEW

**Comment by maryamariyan** [comment]
As per offline discussion with @wtgodbe and based on https://github.com/dotnet/corefx/issues/24357#issuecomment-348608779, in order to fix test failure ```HandleCountChanges``` and also make behavior consistent with netfx, I changed the unix behavior to throw with InvalidOperationException and updated affected tests.

File: src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Unix.cs [hunk 1]
```diff
@@ -74,7 +74,7 @@ internal static void ThrowIfRemoteMachine(string machineName)
         {
             if (IsRemoteMachine(machineName))
             {
-                throw new PlatformNotSupportedException(SR.RemoteMachinesNotSupported);
+                throw new InvalidOperationException(SR.CouldntConnectToRemoteMachine);
```
Comment (stephentoub) [inline src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Unix.cs]: Do we really want to change this one?  It's an unsupported feature on Unix; throwing saying we couldn't connect suggests we tried.

**Comment by stephentoub** [comment]
> comment in #24357 (comment)

My comment was about Windows.  I don't think we should change the Unix exception away from PlatformNotSupportedException.

File: src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs [hunk 2]
```diff
@@ -445,6 +445,6 @@ private void RunTestAsSudo(Func<string, int> testMethod, string arg)
         [DllImport("libc")]
         private static extern int seteuid(uint euid);
 
-        private readonly string[] s_allowedProgramsToRun = new string[] { "xdg-open", "gnome-open", "kfmclient" };
+        private static readonly string[] s_allowedProgramsToRun = new string[] { "xdg-open", "gnome-open", "kfmclient" };
```
Comment (stephentoub) [inline src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs]: This fix LGTM.

File: src/System.Diagnostics.Process/tests/ProcessTests.cs [hunk 3]
```diff
@@ -923,10 +923,10 @@ public void GetProcesses_EmptyMachineName_ThrowsArgumentException()
 
         [Fact]
         [ActiveIssue(26720, TargetFrameworkMonikers.Uap)]
-        [ActiveIssue(27459)]
         public void GetProcesses_InvalidMachineName_ThrowsInvalidOperationException()
         {
-            Assert.Throws<InvalidOperationException>(() => Process.GetProcesses(Guid.NewGuid().ToString()));
+            Type exceptionType = PlatformDetection.IsWindows ? typeof(InvalidOperationException) : typeof(PlatformNotSupportedException);
+            var exception = Assert.Throws(exceptionType, () => Process.GetProcesses(Guid.NewGuid().ToString()));
```
Comment (stephentoub) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: Nit: the `var exception =` isn't needed

Comment (maryamariyan) [reply src/System.Diagnostics.Process/tests/ProcessTests.cs]: Thanks. I'm removing this variable in next PR here: https://github.com/dotnet/corefx/pull/27557

**Pull request merged**